### PR TITLE
Rl gpu

### PR DESCRIPTION
**Pull request by ysqyang**
# Description

1. Removed all device parameters from trainer / train_ops;
2. Moved device management logic to training manager and env sampler level;
3. Provided interfaces (rollout_device_allocator & training_device_allocator) for users to define centralized device allocation;
4. Fixed "get_own_ip_address" bug

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: maro/rl/utils/torch_utils.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def match_shape(tensor: Union[torch.Tensor, np.ndarray], shape: tuple) -> bool:
         return True
 
 
-def ndarray_to_tensor(array: np.ndarray, device: torch.device) -> torch.Tensor:
+def ndarray_to_tensor(array: np.ndarray, device: torch.device = None) -> torch.Tensor:
```
Comment (lihuoran) [inline maro/rl/utils/torch_utils.py:34]: Is it okay to call `.to(device)` is `device` is None?

File: maro/rl/training/train_ops.py [hunk 2]
```diff
@@ -108,6 +101,9 @@ def set_policy_state(self, policy_state: object) -> None:
         """
         self._policy.set_state(policy_state)
 
+    def to_device(self, device: str):
```
Comment (lihuoran) [inline maro/rl/training/train_ops.py:104]: Add return value.

File: maro/rl/training/train_ops.py [hunk 3]
```diff
@@ -108,6 +101,9 @@ def set_policy_state(self, policy_state: object) -> None:
...
+        pass
```
Comment (lihuoran) [inline maro/rl/training/train_ops.py:105]: I think this method should be implemented by all its subclasses. Therefore, we should add `@abstractmethod` and use `raise NotImplementedError`.

File: maro/rl/utils/torch_utils.py (hunk 1 above)
Comment (ysqyang) [reply maro/rl/utils/torch_utils.py:34]: Yes, .to(None) is equivalent to a no-op

**Pull request merged**